internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -126,31 +134,59 @@ func (hr *HypershiftAgentServiceConfigReconciler) Reconcile(origCtx context.Cont
 	}
 
 	// Ensure agent-install CRDs exist on spoke cluster and clean stale ones
-	if err := hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
+	if err = hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
 		log.WithError(err).Error(fmt.Sprintf("Failed to sync agent-install CRDs on spoke cluster in namespace '%s'", instance.Namespace))
 		return ctrl.Result{}, err
 	}
 
 	// Initialize ASC to reconcile components according to instance context
 	asc.initHASC(hr, instance)
 
+	// Ensure relevant finalizers exist (cleanup on deletion)
+	if err = ensureFinalizers(ctx, log, asc, hypershiftAgentServiceConfigFinalizerName); err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if !instance.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	// Invoke validation funcs
+	valid, err = validate(ctx, log, asc)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if !valid {
+		return ctrl.Result{}, nil
+	}
+
+	// Remove IPXE HTTP routes if not needed
+	if err = cleanHTTPRoute(ctx, log, asc); err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+
 	// Reconcile hub components
-	if result, err := hr.reconcileHubComponents(ctx, log, asc); err != nil {
+	if result, err = hr.reconcileHubComponents(ctx, log, asc); err != nil {
 		return result, err
 	}
 
+	// Reconcile image-service stateful-set
+	if err = reconcileImageServiceStatefulSet(ctx, log, asc); err != nil {
+		// TODO: update condition
+		return ctrl.Result{}, err
+	}
+
 	// Switch to spoke client in the reconciler context
 	// TODO: extract client from AgentServiceConfigReconcileContext to avoid sharing
 	//       it with both hub and spoke clients.
 	asc.rec.Client = spokeClient
 
 	// Ensure instance namespace exists on spoke cluster
-	if err := hr.ensureSpokeNamespace(ctx, log, asc); err != nil {
```

Comment:
This should be done on the start. there is no point in continuing the reconciliation of there is no namespace on the spoke side. It should be a pre-condition for the entire process

Reply:
The ensure func creates the namespace if missing, so it should be fine..